### ISSUE TITLE
Simplify ChildView logic

### ISSUE THREAD

**Issue by jmcarp** [eregs/regulations-site#289]
Here's how we open a new section at the moment:

* A `section:open` event in the `MainEvents` namespace is triggered
* The `MainView` singleton fires its `createView` callback
* The appropriate model and view types are instantiated. The `options` object passed to the view passes the `render` method of the main view as `render`.
* In the base `ChildView` constructor, define a callback that calls the `render` method of the `ChildView`, as well as the `render` callback passed through from `MainView`.
* From the `ChildView` constructor, call the `get` method of the attached model, passing through the callback.
* The model builds the url to fetch the html partial and sends an ajax request.
* On ajax success, call the callback, which in turn calls the `render` methods of both `ChildView` and `MainView`.
* From the `render` method of `MainView`, update the html with the returned partial, then trigger `section:rendered`.
* On `section:rendered`, set the element of the `ChildView`.

This is kind of hard to follow: there are lots of steps, multiple callbacks with the same name, and we have to pass callbacks from place to place. And I didn't even describe all the steps! I think this can be simpler, maybe like this:

* A `section:open` event in the `MainEvents` namespace is triggered.
* The `MainView` singleton fires its `createView` callback.
* The appropriate model is instantiated, and its `get` method is called.
* On success, update the html, then instantiate the appropriate view.

I'd like to give this a try and see what happens. Meanwhile, I'm curious to know whether @theresaanna and @ascott1 have thoughts about this direction--is all this indirection necessary? Am I missing something about the way routing works? To be honest, I have a hard time keeping the whole flow in my head at once, so it's very possible that I am!

cc @xtine @cmc333333

**Comment by theresaanna** [comment]
@jmcarp That all sounds correct, though my memory is quite faded. I'm glad this is getting simplified. 

Part of what you see reflected is that we built eRegs up against a couple existential deadlines and my own spaghetti brain. It clearly needed a refactoring pass that it didn't get, though the scary thing is that I did manage to do some refactoring.

I'm really glad to see things getting an overhaul!
